### ISSUE TITLE
<CR> mapping change

### ISSUE THREAD

**Issue by Rich-YoYo** [vimwiki/vimwiki#813]
I found that a mapping defined in vimwiki was interfering with use of the popup/completion menu. The following change fixes it for the completion menu, but as I'm not quite sure what it's supposed to do in vimwiki, I can't be sure I've not broken the existing mapping. Anyway, it's at line 570 in ./ftplugin/vimwiki.vim;-

```
if !hasmapto('VimwikiReturn', 'i')
  if maparg('<CR>', 'i') !~? '<Esc>:VimwikiReturn'
    "inoremap <silent><buffer> <CR> <Esc>:VimwikiReturn 1 5<CR>
    " ### I CHANGED THE ABOVE LINE TO THIS... ####
    " Avoid interfering with popup/completion menu if it's active
    inoremap <expr><silent><buffer> <CR> pumvisible() ? "\<CR>" : "\<Esc>:VimwikiReturn 1 5\<CR>"
  endif
  if maparg('<S-CR>', 'i') !~? '<Esc>:VimwikiReturn'
    inoremap <silent><buffer> <S-CR> <Esc>:VimwikiReturn 2 2<CR>
  endif
endif
```

**Comment by Rich-YoYo** [comment]
While I understand what you're saying (and it's not my project, so who am I to complain?), the problem I found is interfering with base vim functionality - ie, popup auto-complete menus like spellcheck. It's not that it's clashing with some other plugin or my my own random stuff. As such, it looks to me that the mapping should try and avoid the problem. As it stands, this mapping breaks every single vim instance; not just my particular configuration, so it seems "wrong" to have to define an external mapping to get round the problem, especially as that external mapping would have to be aware of internal vimwiki workings (in order to set the mapping to correctly point to 'VimWikiReturn')

**Comment by ranebrown** [comment]
I was thinking it was related to another completion plugin. I agree we shouldn't break default vim behavior if possible.

**Comment by tinmarino** [comment]
@Rich-YoYo  can you make a PR ?

**Comment by Rich-YoYo** [comment]
I'm sorry, github (indeed, git) is a bit of a mystery to me; it seems to work counter to any other vcs I've ever used - I'm not even sure what a "pull request" is. I don't think I can contribute anything beyond the suggested patch that I posted (which may or may not be correct?)

**Comment by tinmarino** [comment]
@Rich-YoYo fixed by: e4186adc3df6c (or last commit in dev referring this issue)

Thank you for the commented path it worked like charm and I learned the `imap <expr>` style which is perfect for this case.

Good job !
